libkrun/libkruntime.c

```diff
@@ -29,15 +29,70 @@
 #define ACTUAL_CLOCK_MONOTONIC    CLOCK_MONOTONIC
 #endif
 
+#define CORE_BOUNDS_CHECK(C)                                                \
```

Comment:
I can make them functions. I think I was trying to ensure no extra calls in the timed section, but it's probably negotiable.